### ISSUE TITLE
[Bug]: [kafka] Search id is wrong after compaction and deletion

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#17178]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: latest
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus==2.1.0.dev61
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search id is wrong after compaction and deletion

```
[2022-05-23T19:07:15.114Z] [gw1] [ 17%] FAILED testcases/test_compaction.py::TestCompactionOperation::test_compact_search_after_delete_channel 
```

### Expected Behavior

Search id is right after compaction and deletion

### Steps To Reproduce

```markdown
Nightly:

https://ci.milvus.io:18080/jenkins/blue/rest/organizations/jenkins/pipelines/milvus-nightly-ci/branches/master/runs/548/nodes/64/steps/177/log/?start=0
```

### Milvus Log

1. milvus log:
[artifacts-milvus-distributed-kafka-master-548-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/548/artifact/artifacts-milvus-distributed-kafka-master-548-pymilvus-e2e-logs.tar.gz)

2. Failed case and timeline:
[2022-05-23T19:07:15.114Z] [gw1] [ 17%] FAILED testcases/test_compaction.py::TestCompactionOperation::test_compact_search_after_delete_channel 

3. collection name: compact_DAsE9Q3Q

4. test case:
```python
    def test_compact_search_after_delete_channel(self):
        """
        target: test search after compact, and queryNode get delete request from channel,
                rather than compacted delta log
        method: 1.insert, flush and load
                2.delete half
                3.compact
                4.search
        expected: No compact, compact get delta log from storage
        """
        collection_w = self.init_collection_wrap(cf.gen_unique_str(prefix), shards_num=1)

        df = cf.gen_default_dataframe_data()
        insert_res, _ = collection_w.insert(df)
        assert collection_w.num_entities == ct.default_nb

        collection_w.load()

        expr = f'{ct.default_int64_field_name} in {insert_res.primary_keys[:ct.default_nb // 2]}'
        collection_w.delete(expr)

        collection_w.compact()
        c_plans = collection_w.get_compaction_plans()[0]
        assert len(c_plans.plans) == 0

        # search
        collection_w.load()
        search_res, _ = collection_w.search(cf.gen_vectors(ct.default_nq, ct.default_dim),
                                            ct.default_float_vec_field_name,
                                            ct.default_search_params, ct.default_limit,
                                            check_task=CheckTasks.check_search_results,
                                            check_items={"nq": ct.default_nq,
                                                         "ids": insert_res.primary_keys[ct.default_nb // 2:],
                                                         "limit": ct.default_limit}
                                            )
```

**Comment by xiaofan-luan** [comment]
so it's saying that after compaction search result distance is ok but id is incorrect?

**Comment by binbinlv** [comment]
The error here says some searched ids which has been deleted, but it still could be searched as return. 

I think here the root cause may be "delete issue" too. 

I open this issue just for confirmation because the phenomenon is a little different in case of different reason. But actually, this case also includes delete operation.

Thanks.

**Comment by jaime0815** [comment]
the root cause is similar to xige description within #16961.

**Comment by longjiquan** [comment]
In this case, insert -> flush -> load -> delete (after load) -> compact -> search. delete will be received by delta channel. However, it seems that no querynode received it. Inverstigating why...

collection id: 433411281367597057
```log
49055:[2022/05/23 19:07:09.970 +00:00] [DEBUG] [task.go:139] ["collection name -> id"] ["collection name"=compact_DAsE9Q3Q] [collection_id=433411281367597057] ["default partition id"=433411281367597058]
```

segment id: 433411281433133058
```log
[2022/05/23 19:07:13.365 +00:00] [INFO] [segment.go:192] ["create segment"] [collectionID=433411281367597057] [partitionID=433411281367597058] [segmentID=433411281433133058] [segmentType=3]
```

**Comment by longjiquan** [comment]
Data node received delete requests, and it should forwad this to delta channel.

```log
61799:[2022/05/23 19:07:14.263 +00:00] [DEBUG] [flow_graph_dd_node.go:150] ["DDNode receive delete messages"] [num=1500] [vChannelName=by-dev-rootcoord-dml_58_433411281367597057v0]
```

**Comment by longjiquan** [comment]
Can be reproduced on my local pc. Investigating why ...

**Comment by xiaofan-luan** [comment]
@longjiquan nice catch! let me know if 17398 is ready for review

**Comment by longjiquan** [comment]
> @longjiquan nice catch! let me know if 17398 is ready for review

Sure, and I think #17398 is ready now.

**Comment by xiaofan-luan** [comment]
@longjiquan could be fixed by #17433, could you help on verify it?

**Comment by longjiquan** [comment]
> @longjiquan could be fixed by #17433, could you help on verify it?

OK, verified on my local pc. #17433 works.

**Comment by binbinlv** [comment]
It has not been reproduced for several days, so fixed and close it.
